### DISCUSSION TITLE
four digit code

### DISCUSSION THREAD

**Discussion by ghost** [community/community#33688] [Other Feature Feedback, Questions, & Ideas]
The verification codes are incessent; everytime I login to Github, I also have to login to my email address just to "verify my device". 
We live in an age where Google and Microsoft follow us around, like peeping Tom's, so if you are counting on cookies to "idenfity devices" then its never going to be user-friendly. Most people delete that when they close their browsers.
Instead of pretending to know what is best for everyone else, like a pre-enlightenment dictator, why not give people an option to turn that feature off?

**Comment by Matteo-it** [comment]
Because they want your data!

**Comment by dragos-tudor** [comment]
GitHub ask me the verification code daily. I don't have 2FA configured so my question is "Why GitHub made decisions behalf of me?".
The verification code policy seems to be based at least on client IP.
I daily have allocated a new IP (by the internet provider DHCP) because my router is turned off over night so I daily have to deal with this extremely annoying behavior.
